homeassistant/components/crownstone/config_flow.py

```diff
@@ -0,0 +1,229 @@
+from crownstone_cloud import CrownstoneCloud
+from crownstone_cloud.exceptions import (
+    CrownstoneAuthenticationError,
+    CrownstoneUnknownError,
+)
+import serial.tools.list_ports
+from serial.tools.list_ports_common import ListPortInfo
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry, ConfigFlow, OptionsFlow
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, CONF_UNIQUE_ID
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import aiohttp_client
+
+from .const import (
+    CONF_USB_MANUAL_PATH,
+    CONF_USB_PATH,
+    CONF_USE_CROWNSTONE_USB,
+    DOMAIN,
+    DONT_USE_USB,
+    MANUAL_PATH,
+    REFRESH_LIST,
+)
+from .helpers import get_serial_by_id
+
+
+class CrownstoneConfigFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Crownstone."""
+
+    VERSION = 1
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> CrownstoneOptionsFlowHandler:
+        """Return the Crownstone options."""
+        return CrownstoneOptionsFlowHandler(config_entry)
+
+    def __init__(self) -> None:
+        """Initialize the flow."""
+        self.login_info: dict[str, Any] = {}
+        self.usb_path: str | None = None
+        self.existing_entry_id: str | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=vol.Schema(
+                    {vol.Required(CONF_EMAIL): str, vol.Required(CONF_PASSWORD): str}
+                ),
+            )
+
+        cloud = CrownstoneCloud(
+            email=user_input[CONF_EMAIL],
+            password=user_input[CONF_PASSWORD],
+            clientsession=aiohttp_client.async_get_clientsession(self.hass),
+        )
+
+        try:
+            # get all cloud data for this account
+            await cloud.async_initialize()
+        except CrownstoneAuthenticationError as auth_error:
+            if auth_error.type == "LOGIN_FAILED":
+                errors["base"] = "invalid_auth"
+            elif auth_error.type == "LOGIN_FAILED_EMAIL_NOT_VERIFIED":
+                errors["base"] = "account_not_verified"
+        except CrownstoneUnknownError:
+            errors["base"] = "unknown_error"
+
+        # show form again, with the errors
+        if errors:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=vol.Schema(
+                    {vol.Required(CONF_EMAIL): str, vol.Required(CONF_PASSWORD): str}
+                ),
+                errors=errors,
+            )
+
+        # set unique id and abort if already configured
+        await self.async_set_unique_id(user_input[CONF_EMAIL])
```

Comment:
Is there an account id that can be used instead?
https://developers.home-assistant.io/docs/config_entries_config_flow_handler/#sometimes-acceptable-sources-for-a-unique-id-for-cloud-services

Reply:
yes there is, good catch